tensorflow/python/kernel_tests/math_ops/reduction_ops_test.py

```diff
@@ -525,10 +571,11 @@ def testFloat32(self):
       self._compareAllAxes(np_arr)
 
   @test_util.run_deprecated_v1
-  def testBfloat16(self):
-    for rank in range(1, _MAX_RANK + 1):
-      np_arr = self._makeIncremental((2,) * rank, dtypes.bfloat16)
-      self._compareAllAxes(np_arr, rtol=1e-3, atol=1.)
+  def testBFloat16(self):
+    with ops.device("/cpu:0"):
```

Comment:
Don't do explicit device placement.  Why doesn't this work on GPU?

Reply:
Changes updated.